app/main.py

```
@@ -1,5 +1,79 @@
+    ) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords is not None else [0, 0]
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None
+    ) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
+        super().__init__(name, weight, coords)
+
+    def go_up(self, step: int = 1) -> None:
+        self.coords[2] += step
+
+    def go_down(self, step: int = 1) -> None:
+        self.coords[2] -= step
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None,
+            max_load_weight: int = 0,
+            current_load: Optional["Cargo"] = None
+    ) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, cargo: "Cargo") -> None:
+        if self.current_load is None and cargo.weight <= self.max_load_weight:
+            self.current_load = cargo
+            print(f"Cargo loaded: {cargo.weight}kg")
+        else:
+            print(f"Cannot load cargo: {cargo.weight}kg")
+
+    def unhook_load(self) -> None:
+        if self.current_load is not None:
+            print(f"Cargo unloaded: {self.current_load.weight}kg")
+            self.current_load = None
+        else:
+            print("No cargo to unload")
```

The same comment for the `unhook_load` method. Instead of printing a message when there is no cargo to unload, it would be better to raise an exception. This would make your code more robust and easier to use in other parts of your program, or even in other programs.